modules/apps/object/object-api/src/main/java/com/liferay/object/exception/RequiredObjectRelationshipException.java

```diff
@@ -7,26 +7,40 @@
 
 import com.liferay.portal.kernel.exception.PortalException;
 
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * @author Marco Leo
  */
 public class RequiredObjectRelationshipException extends PortalException {
 
-	public RequiredObjectRelationshipException() {
-	}
-
-	public RequiredObjectRelationshipException(String msg) {
-		super(msg);
+	public RequiredObjectRelationshipException(String message) {
+		super(message);
 	}
 
 	public RequiredObjectRelationshipException(
-		String msg, Throwable throwable) {
+		String message, String objectDefinition2Name,
+		String objectRelationshipName) {
+
+		super(message);
 
-		super(msg, throwable);
+		_arguments = Arrays.asList(
+			objectRelationshipName, objectDefinition2Name);
+		_messageKey =
+			"the-prevent-deletion-type-in-the-relationship-x-with-object-" +
+				"definition-x-is-preventing-this-entry-from-being-deleted";
 	}
 
-	public RequiredObjectRelationshipException(Throwable throwable) {
-		super(throwable);
+	public List<Object> getArguments() {
+		return _arguments;
 	}
 
+	public String getMessageKey() {
+		return _messageKey;
+	}
+
+	private List<Object> _arguments;
+	private String _messageKey;
+
```

Comment:
Hey @murilostodolni let's use only one constructor for this class that receives an ObjectRelationship, this way we will be able to create the message `Object relationship 54321 does not allow deletes` getting the objectRelationshipId from ObjectRelationship and we will be able to get the arguments for the message key as well, objectRelationshipName from ObjectRelationship and the objectDefinition2Name by using the `ObjectDefinitionLocalServiceUtil.getObjectDefinition`